frontend/src/app/core/state/resource-store.service.ts

```diff
@@ -87,42 +92,61 @@ export abstract class ResourceCollectionService<T extends { id:ID }> {
    * Require the results for the given filter params
    * Returns a cached set if it was loaded already.
    *
-   * @param params List params to require
+   * @param input List params to require, or href of the resource
    * @private
    */
-  public require(params:ApiV3ListParameters):Observable<T[]> {
-    const key = collectionKey(params);
-    if (this.collectionExists(key) || this.collectionLoading(key)) {
-      return this.loadedCollection(key);
+  public requireCollection(input:ResourceKeyInput):Observable<T[]> {
+    const href = this.buildResourceLink(input);
+    if (this.collectionExists(href) || this.resourceLoading(href)) {
+      return this.loadedCollection(href);
     }
 
     return this
-      .fetchCollection(params)
+      .fetchCollection(href)
       .pipe(
-        switchMap(() => this.loadedCollection(key)),
+        switchMap(() => this.loadedCollection(href)),
       );
   }
 
+  /**
+   * Require a single entity to be loaded.
+   * Returnes the cached entity if it was loaded already
+   *
+   * @param href {string}
+   */
+  public requireEntity(href:string):Observable<T> {
```

Comment:
I just merged that branch into my rework of the storage component. and there I have noticed an effect of `requireEntity`. It must be the same as with `requireCollection`, but apparently we do not have a case yet.

So, in `ngOnInit()` I want to initialize my storage data:

```typescript
this.storage = this.storagesResourceService.requireEntity(this.projectStorage._links.storage.href);
```

the, this observable is used in a vast variety of places. In the template, in the component code fetching data before displaying texts, etc.

BUT: The observable behind it is the `fetchEntity()`, not the `lookup()`. Hence, every subscription to the storage observable executes a new HTTP request.

to circumvent it, I would have to write something like this:

```typescript
    this.storagesResourceService.requireEntity(this.projectStorage._links.storage.href).subscribe();
    this.storage = this.storagesResourceService.lookup(idFromLink(this.projectStorage._links.storage.href));
```
But from my POV this is something, that should be here in the resource store service. What are your opinions on that?

Reply:
Yes I agree, the outer interface should allow multiple subscribes transparently. In fact I even think an internal double subscription to the same fetch should be shared. Added that to both entity and collection here: https://github.com/opf/openproject/commit/2d09306a22

Reply:
Perfect!